include/SQLiteCpp/Backup.h

```diff
@@ -114,13 +115,12 @@ class Backup
     int executeStep(const int aNumPage = -1);
 
     /// Return the number of source pages still to be backed up as of the most recent call to executeStep().
-    int getRemainingPageCount();
+    int getRemainingPageCount() const;
 
     /// Return the total number of pages in the source database as of the most recent call to executeStep().
-    int getTotalPageCount();
+    int getTotalPageCount() const;
 
 private:
-    // TODO: use std::unique_ptr with a custom deleter to call sqlite3_backup_finish()
```

Comment:
Hey, sorry for the misunderstanding; I loved the usage of unique ptr, it just didn't need the custom deleter

Reply:
I can restore unique_ptr and define its destructor using lambda (in Backup constructor), if you want.

Reply:
Sorry, I might be missing something, but there is no reason to do anything specific IMO ; just use the unique ptr, reset will call the existing destructor of Backup. Right?
You can also let it assise and I will handle it after if it's not that simple

Reply:
Hey, I have finally merge this as-is and restored your original code with the custom deleted, thansk a lot!